src/unity/python/turicreate/toolkits/activity_classifier/_activity_classifier.py

```diff
@@ -281,12 +282,27 @@ def _load_version(cls, state, version):
             ('weights', (state['_recalibrated_batch_size'], state['_predictions_in_chunk'], 1))
         ]
 
+        from ._model_architecture import _define_model
+        import mxnet as _mx
         context = _mxnet_utils.get_mxnet_context(max_devices=state['num_sessions'])
-        state['_loss_model'] = _mxnet_utils.load_mxnet_model_from_state(
-            state['_loss_model'], data, labels, None, context)
```

Comment:
Doesn't this mean we're no longer **backward** compatible?
If someone saved a model using version 1, the weights are now saved only in the loss model, and therefore when   we later in lines 301-303 load them from `state['_pred_model']` they would be all zeros, won't they?

I can understand not being forward compatible (saved in new version should not load in old version). But backwards compatibility is important.
We could check for `if cls._PYTHON_ACTIVITY_CLASSIFIER_VERSION == 1 or '_loss_model' in state` then extract the params from loss model. Else extract from pred model.
Right?

Reply:
Thanks for the review! In the current v4, there is no weight sharing when it gets saved to file. All weights are saved twice. Looking at the actualy saved files, a model saved with v4 takes 4 MB while a model saved with v4+ takes 2 MB. Therefore, there is no problem for v4+ to simply ignore half of those weights and load the model entirely from the `pred_model`.

Also, regarding backward compatibility. Every cell in the 6x6 matrix I showed in the original post is the result of an **actual test** and not just my hopes (I wanted to be very thorough!), so I have tested and verified full backward compatibility.